exporters/deploytime/app.py

```diff
@@ -50,54 +52,74 @@ def image_sha(img_url):
-                if i is not None:
-                    metric = DeployTimeMetric(rc.metadata.name, namespace)
-                    metric.labels = rc.metadata.labels
-                    metric.deploy_time = rc.metadata.creationTimestamp
-                    metric.image_sha = i
-                    metrics.append(metric)
+        logging.info("Wwtching namespaces %s", namespaces)
+
+    pods = v1_pods.get(label_selector=pelorus.get_app_label(), field_selector='status.phase=Running').items
+
+    replicas_dict = {}
+    # Process ReplicationControllers for DeploymentConfigs
+    get_replicas('v1', 'ReplicationController', replicas_dict)
+
+    # Process ReplicaSets from apps/v1 api version for Deployments
+    get_replicas('apps/v1', 'ReplicaSet', replicas_dict)
+
+    # Process ReplicaSets from extentions/v1beta1 api version for Deployments
+    get_replicas('extensions/v1beta1', 'ReplicaSet', replicas_dict)
+
+    for pod in pods:
+        if not namespaces or (pod.metadata.namespace in namespaces):
+            logging.info("Getting Replicas for pod: %s", pod.metadata.name)
+            ownerRefs = pod.metadata.ownerReferences
+            namespace = pod.metadata.namespace
+
+            for ownerRef in ownerRefs:
+                if ownerRef.kind in ['ReplicaSet', 'ReplicationController'] and not pod_replica_dict.get(ownerRef.name):
+
+                    logging.info("Getting Replica ownerRef: %s, %s",  ownerRef.name, ownerRef.kind)
+
+                    # Process ReplicationControllers for DeploymentConfigs
+                    rc = replicas_dict[ownerRef.name]
+
+                    if rc:
+                        pod_replica_dict[rc.metadata.name] = "DONE"
+                        images = [image_sha(c.image) for c in pod.spec.containers]
+
+                        # Since a commit will be built into a particular image and there could be multiple
+                        # containers (images) per pod, we will push one metric per image/container in the
+                        # pod template
+                        for i in images:
+                            if i is not None:
+                                metric = DeployTimeMetric(rc.metadata.name, namespace)
+                                metric.labels = rc.metadata.labels
+                                metric.deploy_time = rc.metadata.creationTimestamp
+                                metric.image_sha = i
+                                metric.namespace = namespace
+                                metrics.append(metric)
 
     return metrics
 
 
+def get_replicas(apiVersion, objectName, replicas):
+    # Process ReplicaSets from apps/v1 api version for Deployments
+    try:
+        apiResource = dyn_client.resources.get(api_version=apiVersion, kind=objectName)
+        replicationobjects = apiResource.get(label_selector=pelorus.get_app_label())
+        for replica in replicationobjects.items:
+            replicas[replica.metadata.name] = replica
```

Comment:
using name only will not guarantee uniqueness. I would recommend combining `namespace/name`